File: php/features.php
<?php
declare(strict_types=1);

// Argentofx SDK feature factory

require_once __DIR__ . '/feature/BaseFeature.php';
require_once __DIR__ . '/feature/TestFeature.php';


class ArgentofxFeatures
{
    public static function make_feature(string $name)
    {
        switch ($name) {
            case "base":
                return new ArgentofxBaseFeature();
            case "test":
                return new ArgentofxTestFeature();
            default:
                return new ArgentofxBaseFeature();
        }
    }
}

File: py/features.py
# Argentofx SDK feature factory

from feature.base_feature import ArgentofxBaseFeature
from feature.test_feature import ArgentofxTestFeature


def _make_feature(name):
    features = {
        "base": lambda: ArgentofxBaseFeature(),
        "test": lambda: ArgentofxTestFeature(),
    }
    factory = features.get(name)
    if factory is not None:
        return factory()
    return features["base"]()
